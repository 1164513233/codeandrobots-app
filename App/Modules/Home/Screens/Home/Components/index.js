import React, { Component } from 'react'
import { StyleSheet, View, Image, Text } from 'react-native'
import PropTypes from 'prop-types'

import TouchableOpacity from 'App/Components/TouchableOpacity'

import { ApplicationStyles, Images } from 'App/Themes'

const styles = StyleSheet.create({
  ...ApplicationStyles.screen
})

export default class Screen extends Component {
  static propTypes = {
    onLearnMorePress: PropTypes.func.isRequired
  }

  render () {
    const { onLearnMorePress } = this.props
    return (
      <View style={styles.mainContainer}>
        <View style={styles.helloImage}>
          <Image source={Images.Hello} />
        </View>
        <View style={styles.footer}>
          <View style={[styles.footerSection, styles.footerTitle]}>
            <Text style={styles.text}>HELLO</Text>
          </View>
          <View style={[styles.footerSection, styles.footerSubtitle]}>
            <Text style={styles.footerSubtitleText}>{'The app is ready! Sorry, almost ready :)\n\nThe good news is you can follow along as this Beta version of the app develops over the coming weeks.'}</Text>
          </View>
          <View style={styles.footerSection}>
            <TouchableOpacity style={styles.button} onPress={onLearnMorePress}>
              <Text style={styles.buttonText}>Learn More</Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    )
  }
}
