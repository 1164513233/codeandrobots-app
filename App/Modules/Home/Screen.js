import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { isIOSProduction } from 'App/Services/Properties'

import { Container, List, ListItem, BottomNav } from 'App/Components'

export default class Screen extends Component {
  static propTypes = {
    onNavigatePress: PropTypes.func.isRequired
  }

  render () {
    const { onNavigatePress } = this.props
    return (
      <Container>
        {!isIOSProduction() &&
          <List>
            <ListItem
              title='Play & Explore'
              text='Let’s play'
              button='Play'
              onPress={() => { onNavigatePress('PlayerScreen') }} />
            <ListItem
              title='Code Lab'
              text='Let’s see what you can do'
              button='Code'
              onPress={() => { onNavigatePress('CodeLabScreen') }} />
            <ListItem
              title='Learn'
              text='You’ll be a guru in no time'
              button='Learn'
              onPress={() => { onNavigatePress('LearnScreen') }} />
            <ListItem
              title='Lab'
              text='Get under the hood'
              button='Lab'
              onPress={() => { onNavigatePress('LabScreen') }} />
          </List>
        }
        {isIOSProduction() &&
          <List>
            <ListItem
              title='Play & Explore'
              text='Let’s play'
              button='Play'
              onPress={() => { onNavigatePress('PlayScreen') }} />
            <ListItem
              title='Discover'
              text='Let’s see what you can do'
              button='Discover'
              onPress={() => { onNavigatePress('DiscoverScreen') }} />
            <ListItem
              title='Learn'
              text='You’ll be a guru in no time'
              button='Learn'
              onPress={() => { onNavigatePress('LearnScreen') }} />
          </List>
        }
        <BottomNav
          onRatePress={() => { onNavigatePress('RateScreen') }}
          onHomePress={() => { onNavigatePress('PlayerScreen') }}
          onSettingsPress={() => { onNavigatePress('SettingsScreen') }} />
      </Container>
    )
  }
}
