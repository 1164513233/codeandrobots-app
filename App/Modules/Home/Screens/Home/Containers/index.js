import React, { Component } from 'react'
import { connect } from 'react-redux'

import Screen from '../Components'

export class HomeContainer extends Component {
  render () {
    return (
      <Screen
        ref={(ref) => {
          this.screen = ref
        }}
        {...this.props}
      />
    )
  }
}

const mapStateToProps = (state) => {
  return {
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(HomeContainer)
