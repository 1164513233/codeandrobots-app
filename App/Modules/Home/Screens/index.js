
import Home from './Containers'

export {
  Home
}
