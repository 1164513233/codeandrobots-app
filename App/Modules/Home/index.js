
import { Home } from './Screens'

export {
  Home
}
