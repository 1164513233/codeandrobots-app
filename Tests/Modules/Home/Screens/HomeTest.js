import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Home from 'App/Modules/Home/Screens/Components'

describe('Home', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Home onLearnMorePress={() => {}} />)
  })

  it('Component renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
