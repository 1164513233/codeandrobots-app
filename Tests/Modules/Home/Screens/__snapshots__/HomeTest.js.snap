// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders correctly 1`] = `
<Container>
  <Card
    button="Learn More"
    image={
      Object {
        "process": [Function],
      }
    }
    onPress={[Function]}
    text="The app is ready! Sorry, almost ready :)

The good news is you can follow along as this Beta version of the app develops over the coming weeks."
    title="HELLO"
  />
  <BottomNav />
</Container>
`;
